src/Sylius/Bundle/ApiBundle/Validator/Constraints/UserAlreadyRegisteredValidator.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\Constraints;
+
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Sylius\Component\Core\Repository\CustomerRepositoryInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+/** @experimental  */
+final class UserAlreadyRegisteredValidator extends ConstraintValidator
+{
+    public function __construct(
+        private CustomerRepositoryInterface $customerRepository,
+        private TokenStorageInterface $tokenStorage
+    ) {
+    }
+
+    public function validate($value, Constraint $constraint): void
+    {
+        /** @var UserAlreadyRegistered $constraint */
+        Assert::isInstanceOf($constraint, UserAlreadyRegistered::class);
+
+        $token = $this->tokenStorage->getToken();
+
+        /** @var CustomerInterface|null $existingCustomer */
+        $existingCustomer = $this->customerRepository->findOneBy(['email' => $value->getEmail()]);
+        if (null !== $existingCustomer && !$token->getUser() instanceof ShopUserInterface) {
```

Comment:
There an issue here, checking that token is just there, will be not enough cause it may be different customer sending different email:
- email for current token is abc@efg.com,
- email sent in checkout is hij@efg.com,
- current check will pass if there is customer with second email, even if the one in token is the first one

Reply:
Yup you are right but if the customer is logged then he can't override his email and even if he tries to override it we still get his email from an already logged user so in this case checking it this way is enough 😃